blocks/library/text/index.js

```diff
@@ -48,40 +45,46 @@ registerBlock( 'core/text', {
 	],
 
 	edit( { attributes, setAttributes, insertBlockAfter } ) {
-		const { content, align } = attributes;
+		const { content = <p />, align } = attributes;
 
 		return (
 			<Editable
-				value={ fromValueToParagraphs( content ) }
-				onChange={ ( paragraphs ) => setAttributes( {
-					content: fromParagraphsToValue( paragraphs )
-				} ) }
+				value={ content }
+				onChange={ ( nextContent ) => {
+					setAttributes( {
+						content: nextContent
+					} );
+				} }
 				style={ align ? { textAlign: align } : null }
 				onSplit={ ( before, after ) => {
-					setAttributes( { content: fromParagraphsToValue( before ) } );
+					setAttributes( { content: before } );
 					insertBlockAfter( wp.blocks.createBlock( 'core/text', {
-						content: fromParagraphsToValue( after )
+						content: after
 					} ) );
 				} }
 			/>
 		);
 	},
 
 	save( { attributes } ) {
-		const { align, content } = attributes;
+		let { content } = attributes;
+		if ( ! Array.isArray( content ) ) {
+			content = [ content ];
+		}
 
-		// Todo: Remove the temporary <div> wrapper once the serializer supports returning an array
-		return (
-			<div>
-				{ content && content.map( ( paragraph, i ) => (
-					<p
-						key={ i }
-						style={ align ? { textAlign: align } : null }
-						dangerouslySetInnerHTML={ {
-							__html: paragraph
-						} } />
-				) ) }
-			</div>
-		);
+		const { align } = attributes;
+		if ( ! align ) {
+			return content;
+		}
+
+		return content.map( ( paragraph ) => {
+			if ( 'string' === typeof paragraph ) {
```

Comment:
>When do we have a string here?

When splitting a paragraph into two, the default TinyMCE `getContent` was including a newline between the paragraphs which when parsed by `html-to-react` resulted in an `[ Element, '\n', Element ]`. In the rebased 075237ef5d3947a9f784c27580bf6ba9a9193a4e, I changed this to get content as raw formatted with the assumption that clean-up will occur as a result of `html-to-react` and `renderToString` (see also extended commit description).